tests/v2/test_1135-rpad-operation.py

```diff
@@ -0,0 +1,1297 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
+import pytest  # noqa: F401
+import numpy as np  # noqa: F401
+import awkward as ak  # noqa: F401
+from awkward._v2.tmp_for_testing import v1_to_v2  # noqa: F401
+
+pytestmark = pytest.mark.skipif(
+    ak._util.py27, reason="No Python 2.7 support in Awkward 2.x"
+)
+
+
+def test_rpad_and_clip_empty_array():
+    empty = ak._v2.contents.emptyarray.EmptyArray()
+
+    assert ak.to_list(empty) == []
+    assert ak.to_list(empty.rpad(5, 0)) == [None, None, None, None, None]
+    assert ak.to_list(empty.rpad_and_clip(5, 0)) == [None, None, None, None, None]
+
+
+def test_rpad_and_clip_numpy_array():
+    array = ak._v2.contents.numpyarray.NumpyArray(
+        np.arange(2 * 3 * 5, dtype=np.int64).reshape(2, 3, 5)
+    )
+    assert ak.to_list(array) == [
+        [[0, 1, 2, 3, 4], [5, 6, 7, 8, 9], [10, 11, 12, 13, 14]],
+        [[15, 16, 17, 18, 19], [20, 21, 22, 23, 24], [25, 26, 27, 28, 29]],
+    ]
+
+    assert ak.to_list(array.rpad_and_clip(5, 0)) == [
+        [[0, 1, 2, 3, 4], [5, 6, 7, 8, 9], [10, 11, 12, 13, 14]],
+        [[15, 16, 17, 18, 19], [20, 21, 22, 23, 24], [25, 26, 27, 28, 29]],
+        None,
+        None,
+        None,
+    ]
```

Comment:
If you're planning on converting all `rpad_and_clip(target, axis)` into `rpad(target, axis, clip=True)`, in preparation for the name change (turning function names into argument values is more awkward than just changing names), know that there's an easier way to do it than doing it all by hand.

In Emacs, this is an ideal candidate for a keyboard macro:

```lisp
M-x start-kbd-macro            ; C-x (
M-x search-forward             ; C-s
.rpad_and_clip(
M-x backward-char              ; C-b or left arrow
M-x delete-backward-char × 9   ; backspace 9 times to delete "_and_clip"
M-x forward-sexp               ; C-M-f to go forward over parenthesized expression
M-x backward-char              ; C-b or left arrow
, clip=True
M-x end-kbd-macro              ; C-x )
```

Then you can run the above keyboard macro by `M-x call-last-kbd-macro` or `C-x e` (and repeatedly by pressing `e` thereafter). It would search forward to all ".rpad_and_clip", erase the "_and_clip", jump to the end of the parenthesized expression (regardless of whatever's in it), and add the ", clip=True". It's the sort of thing that I would never do through `M-x` commands—I'd use the keyboard shortcuts; I'm just being explicit to be understood. Since some of those shortcuts navigate syntactically through the code (jumping over parentheses), it's possible to do sophisticated, syntax-aware edits, not just find-and-replace. Since it plays them back only once per `C-x e` (or `e`), you get to keep an eye on it and make sure that it's not messing things up—semi-automated can be overall faster than manual _and_ faster than full automation, since a fully automated script can fail to handle unforeseen cases and then you have to go back and fix the mess it's made.

I'm telling you all of this so that we globally optimize our use of time. If you would have done this edit manually or by writing a script, note that it can be done in less than a minute with a keyboard macro. Either I'll do it later, as one of the tasks in a future renaming pass, or you might want to try it in Emacs, or—best yet—you can tell me if a feature like this exists in any modern (i.e. non-Emacs) IDE. I'd love to know. I've been searching for it, wondering how users of modern IDEs get along without having this feature, but maybe it exists by another name.

(Actually, I don't remember what your preferred IDE was. If I'm telling you something you already know; sorry!)

Reply:
Thanks for this! This might be cheating but I am currently using `Neovim` which is a `modern` Vim fork. I followed similar steps to register a macro and then execute it for every instance in one go. This is how I did it in Vim, there could be better ways of doing this, but I am not currently very aware of it (I started shifting my workflow to `Neovim`, only like a week back)
```
qa                          "Start recording a macro and store it in the register 'a'
/.rpad_and_clip(            "Find the first instance of "rpad_and_clip(" from the current line
f_                          "jump to the first "_"
de                          "delete till end of the work i.e _and_clip
%                           "go to the matching closing parenthesis i.e )
<InsertMode>                       
, clip=True                 "Insert the kwarg
<EscToNormalMode>
q                           "Stop recording the macro
```
Then we just do a `@a` for the first time and then just spam `@@` for every subsequent macro execution. It's pretty neat. For like proper modern IDEs / Text Editors, it becomes a bit difficult. I am not sure if there's a better way, but in VS Code, you'll have to write some editor specific commands in a JSON object and install an [extension](https://marketplace.visualstudio.com/items?itemName=geddski.macros) to get the macro to work.

Reply:
All right! Then it can be done easily in Emacs and Vim (I'm guessing the above works in normal Vim, not just Neovim). It seems to be the killer app for programmatic text editing, and it's crucial that there's a low barrier to entry (i.e. not having to install an extension, not having to write it out, but "programming by example"). I'm still surprised that it's not ubiquitous among IDEs.